controller/src/controller.rs

```diff
@@ -86,6 +88,32 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
                 )
                 .await
                 .context(error::UpdateNodeSpec)
+        } else if node.has_crashed() {
+            self.emit_metrics()?;
+
+            let desired_spec = determine_next_node_spec(&node);
+
+            event!(
+                Level::INFO,
+                "BottlerocketShadow's status is ErrorReset, try to recover from crash."
+            );
+
+            self.node_client
+                .update_node_spec(
+                    &node.selector().context(error::NodeSelectorCreation)?,
+                    &desired_spec,
+                )
+                .await
+                .context(error::UpdateNodeSpec)?;
+
+            // Rely on active_node_set to determine whether the agent
+            // change status back to Idle
+            while !self.active_node_set().is_empty() {
+                event!(Level::INFO, "Wait for the agent to recover");
+                sleep(AGENT_RECOVERY_INTERVAL).await;
+            }
```

Comment:
In the existing logic that crash_count is in the spec, the controller was in charge of incrementing the crash_count once it sees the node in ErrorReset, so letting it go to another loop would cause false incrementing the crash_count.

>I don't think we should sleep loop here, because it could mean that the controller ceases to emit metrics for the duration of time that this node attempts to recover.

In the previous controller logic, the controller only emits metrics once it detects the node reached the desired state, which means the controller also ceases to emit metrics for the duration of time the node work on state transfer.